### PR TITLE
Feature/core 685 wallet component

### PR DESCRIPTION
**Pull request by Tschuck**

### CODE REVIEW

File: dapps/profile.vue/src/i18n/de.json [hunk 1]
```diff
@@ -130,6 +154,7 @@
       "profile-type": "Identitätstyp wird geändert...",
       "profile-update": "Identitätinformationen werden gespeichert...",
       "request-verification": "Verifikation wird beantragt...",
+      "send-eve": "Sending Eve...",
```
Comment (wulfraem) [inline dapps/profile.vue/src/i18n/de.json]:
```suggestion
      "send-eve": "EVEs werden gesendet...",
```

File: dapps/profile.vue/src/i18n/de.json [hunk 2]
```diff
@@ -388,8 +414,78 @@
...
+          "unknown_state": "Es ist ein Fehler beim Bezahlvorgang aufgetreten. Bitte kontaktieren Sie unseren Kundensupport.",
+          "wallet_not_enough_funds": "Es ist ein Fehler im evan.network aufgetreten. <b> Ihre Karte wurde NICHT belastet! </ b> Bitte versuchen Sie es später erneut"
+        },
+        "payForm": {
+          "amount": {
+            "amount": {
+              "error": "Bitte geben Sie eine Zahl ein!",
+              "error-less": "Sie müssen mindestens 10 EVE kaufen!",
+              "label": "EVE",
+              "placeholder": "Menge an Eve, die sie gekauft werden sollen."
```
Comment (wulfraem) [inline dapps/profile.vue/src/i18n/de.json]:
```suggestion
              "placeholder": "Menge an Eve, die gekauft werden sollen."
```

File: dapps/profile.vue/src/i18n/de.json [hunk 3]
```diff
@@ -388,8 +414,78 @@
...
+          "accountId": {
+            "error": "Bitte geben Sie eine korrekte Adresse ein.",
+            "label": "Identität",
+            "placeholder": "Kontakt oder Identitätsadresse"
+          },
+          "amount": {
+            "error": "Bitte geben Sie eine Zahl ein!",
+            "error-less": "Ihr Kontostand ist zu gering! ({balance} EVE)",
+            "label": "EVE",
+            "placeholder": "Menge an Eve, die sie gesendet werden sollen."
```
Comment (wulfraem) [inline dapps/profile.vue/src/i18n/de.json]:
```suggestion
            "placeholder": "Menge an Eve, die gesendet werden sollen."
```

File: dapps/profile.vue/src/i18n/en.json [hunk 4]
```diff
@@ -38,6 +53,15 @@
           "placeholder": "My street 15"
         },
         "title": "Contact",
+        "vat": {
+          "billing_country_and_vat_missmatch": "Provided VAT number does not match given country code.",
+          "error": "The VAT number could not be validated!",
+          "eu_billing_vat_not_given": "When buying EVEs from outside of Germany and in the European Union a valid VAT number.",
```
Comment (wulfraem) [inline dapps/profile.vue/src/i18n/en.json]:
```suggestion
          "eu_billing_vat_not_given": "When buying EVEs from outside of Germany and in the European Union a valid VAT number is required.",
```

File: dapps/profile.vue/src/i18n/en.json [hunk 5]
```diff
@@ -130,6 +154,7 @@
       "profile-type": "Changing Identity type...",
       "profile-update": "Updating Identity information...",
       "request-verification": "Requesting Verification...",
+      "send-eve": "Sending Eve...",
```
Comment (wulfraem) [inline dapps/profile.vue/src/i18n/en.json]:
```suggestion
      "send-eve": "Sending EVEs...",
```

File: dapps/profile.vue/src/i18n/en.json [hunk 6]
```diff
@@ -388,8 +414,76 @@
...
+        "accept": {
+          "description": "Do you want to continue with sending <b>{amount} EVE</b> to <b>{name}</b>?",
+          "send": "Send",
+          "title": "Send EVE"
+        },
+        "form": {
+          "accountId": {
+            "error": "You must insert a valid address.",
+            "label": "Identity",
+            "placeholder": "Kontakt or identity address"
```
Comment (wulfraem) [inline dapps/profile.vue/src/i18n/en.json]:
```suggestion
            "placeholder": "Contact or identity address"
```

**Review by wulfraem — CHANGES_REQUESTED** [review]

File: dapps/profile.vue/src/components/wallet/buy/BuyEve.ts [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+
+import Component, { mixins } from 'vue-class-component';
+import axios from 'axios';
+
+import { PaymentService } from '../paymentService';
+import { ErrorStatus } from '../interfaces';
+import CompanyContactForm from 'components/profile/company/contact/contact';
+
+// TODO: evan style
+const elementStyles = {
```
Comment (DanielMaass) [inline dapps/profile.vue/src/components/wallet/buy/BuyEve.ts]: This seems a little messy. Is there an integrating solution?

File: dapps/profile.vue/src/components/wallet/paymentService.ts [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+  OptionsInterface,
+  VatValidationInterface,
+  CustomerParams
+} from './interfaces';
+import { StatusResponse } from './StatusResponse.interface';
+import { StripeSource } from './StripeSource.interface';
+import { relativeTimeRounding } from 'moment';
+
+declare var Stripe: any;
+const PUB_KEY = 'pk_test_kpO3T5fXA7aaftg9D0OO0w3S';
```
Comment (DanielMaass) [inline dapps/profile.vue/src/components/wallet/paymentService.ts]: We should this put into a stripe-config.ts

File: dapps/profile.vue/src/components/wallet/paymentService.ts [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+    this.runtime = runtime;
+  }
+
+  /**
+   * Ensure that stripe is loaded.
+   */
+  async ensureStripe() {
+    return new Promise((resolve) => {
+      const stripeScriptId = 'stripeScript';
+      const stripeScriptPath = 'https://js.stripe.com/v3/';
```
Comment (DanielMaass) [inline dapps/profile.vue/src/components/wallet/paymentService.ts]: this is maybe for a stripe-config.ts, as well

File: dapps/profile.vue/src/components/wallet/paymentv3.service.ts [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+
+import { CustomerInterface, CustomerParams } from './interfaces';
+
+/**
+ * TODO: v3 version!
+ *
+ * Backend:
+ * https://github.com/stripe-samples/accept-a-card-payment/blob/master/using-webhooks/server/node/server.js
+ *
+ * Frontend: https://github.com/stripe-samples/accept-a-card-payment/tree/master/using-webhooks/client/web
```
Comment (DanielMaass) [inline dapps/profile.vue/src/components/wallet/paymentv3.service.ts]: should we remove?

File: dapps/profile.vue/src/components/wallet/paymentv3.service.ts [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+ *
+ */
+
+declare let Stripe: any;
+const PUB_KEY = 'pk_test_kpO3T5fXA7aaftg9D0OO0w3S';
```
Comment (DanielMaass) [inline dapps/profile.vue/src/components/wallet/paymentv3.service.ts]: This is probably all invalid if we use a stripe-config.ts!! ???

File: dapps/profile.vue/src/components/wallet/paymentv3.service.ts [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+
+export class PaymentServiceV3 {
+  private agentUrl = null;
+  private stripe;
+  private authorization;
+  private runtime;
+
+  card;
+
+  private elementStyles = {
```
Comment (DanielMaass) [inline dapps/profile.vue/src/components/wallet/paymentv3.service.ts]: same as paymentService.ts

File: dapps/profile.vue/src/components/wallet/paymentv3.service.ts [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+      },
+      {
+        headers: {
+          authorization: await this.getAuthHeaders()
+        }
+      }
+    );
+
+    console.log('pub', publishableKey);
+    console.log('priv', clientSecret);
```
Comment (DanielMaass) [inline dapps/profile.vue/src/components/wallet/paymentv3.service.ts]: remove console logs

File: dapps/profile.vue/src/components/wallet/paymentv3.service.ts [hunk 14]
```diff
@@ -0,0 +1,213 @@
+/*
```
Comment (DanielMaass) [inline dapps/profile.vue/src/components/wallet/paymentv3.service.ts]: maybe we don't should commit this file?

File: dapps/profile.vue/src/i18n/de.json (hunk 1 above)
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/de.json]: ist nicht deutsch ;)

File: dapps/profile.vue/src/i18n/de.json [hunk 15]
```diff
@@ -388,8 +414,78 @@
...
+          "transaction_failed": "Ihre Zahlung bzw. Abbuchung war erfolgreich, aber die Transaktion ist in der Blockchain fehlgeschlagen. <b> Ihre Karte wurde belastet! </ b> Bitte wenden Sie sich an unseren Kundensupport.",
+          "unknown_state": "Es ist ein Fehler beim Bezahlvorgang aufgetreten. Bitte kontaktieren Sie unseren Kundensupport.",
+          "wallet_not_enough_funds": "Es ist ein Fehler im evan.network aufgetreten. <b> Ihre Karte wurde NICHT belastet! </ b> Bitte versuchen Sie es später erneut"
+        },
+        "payForm": {
+          "amount": {
+            "amount": {
+              "error": "Bitte geben Sie eine Zahl ein!",
+              "error-less": "Sie müssen mindestens 10 EVE kaufen!",
+              "label": "EVE",
```
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/de.json]: "Number of EVE (at least 10)" as label

File: dapps/profile.vue/src/i18n/de.json (hunk 2 above)
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/de.json]: "Anzahl der EVE, die Sie kaufen wollen"

File: dapps/profile.vue/src/i18n/de.json (hunk 3 above)
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/de.json]: "Anzahl der EVE, die Sie versenden möchten"

File: dapps/profile.vue/src/i18n/en.json (hunk 5 above)
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/en.json]: ... EVE (maybe search and replace)

File: dapps/profile.vue/src/i18n/en.json [hunk 16]
```diff
@@ -388,8 +414,76 @@
...
+          "too_many_accounts": "An unknown error occurred regarding your account. <b>Your card was NOT debited!</b> Please contact our customer support.",
+          "transaction_failed": "Your payment/charge was successful but the transaction on blockchain failed. <b>Your card was debited!</b> Please contact our customer support.",
+          "unknown_state": "An unknown error occurred during the payment process. Please contact our customer support.",
+          "wallet_not_enough_funds": "An unknown error on the evan.network. <b>Your card was NOT debited!</b> Please try again later."
+        },
+        "payForm": {
+          "amount": {
+            "error": "Please insert a correct number!",
+            "error-less": "You must by at least 10! ({balance} EVE)",
+            "label": "EVE",
```
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/en.json]: Amount of EVE (at least 10)

File: dapps/profile.vue/src/i18n/en.json [hunk 17]
```diff
@@ -388,8 +414,76 @@
...
+            "placeholder": "EVE to buy"
```
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/en.json]: Enter a number

File: dapps/profile.vue/src/i18n/en.json (hunk 6 above)
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/en.json]: Kontakt to Contact

File: dapps/profile.vue/src/i18n/en.json [hunk 18]
```diff
@@ -388,8 +414,76 @@
...
+          "accountId": {
+            "error": "You must insert a valid address.",
+            "label": "Identity",
+            "placeholder": "Kontakt or identity address"
+          },
+          "amount": {
+            "error": "Please insert a correct number!",
+            "error-less": "Account balance is to low! ({balance} EVE)",
+            "label": "EVE",
+            "placeholder": "EVE to send"
```
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/en.json:481]: Enter a number

File: dapps/profile.vue/src/i18n/en.json [hunk 19]
```diff
@@ -388,8 +414,76 @@
...
+        "form": {
+          "accountId": {
+            "error": "You must insert a valid address.",
+            "label": "Identity",
+            "placeholder": "Kontakt or identity address"
+          },
+          "amount": {
+            "error": "Please insert a correct number!",
+            "error-less": "Account balance is to low! ({balance} EVE)",
+            "label": "EVE",
```
Comment (DanielMaass) [inline dapps/profile.vue/src/i18n/en.json]: Amount of EVE

**Review by wulfraem — APPROVED** [review]

**Pull request merged**